pkg/karmadactl/rebalance/rebalance.go

```diff
@@ -0,0 +1,253 @@
+}
+
+// Complete ensures that options are valid and marshals them if necessary
+func (o *CommandRebalanceOption) Complete(f util.Factory, args []string) error {
+	if len(args) != 2 {
+		return fmt.Errorf("incorrect command format, please use correct command format")
+	}
+
+	if o.Namespace == "" {
+		var err error
+		o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+		if err != nil {
+			return fmt.Errorf("failed to get namespace from Factory. error: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Validate checks to the RebalanceOptions to see if there is sufficient information run the command
+func (o *CommandRebalanceOption) Validate() error {
+	return nil
+}
+
+// Run rebalance resources to achieve a fresh rescheduling
+func (o *CommandRebalanceOption) Run(f util.Factory, args []string) error {
+	// 1. get target rebalance object
+	obj, err := o.getObjInfo(f, args)
+	if err != nil {
+		return fmt.Errorf("failed to get resource in Karmada. err: %w", err)
+	}
+
+	// 2. get karmada-apiserver client
+	karmadaClient, err := f.KarmadaClientSet()
+	if err != nil {
+		return fmt.Errorf("failed to get karmada-apiserver client. err: %w", err)
+	}
+
+	// 3. rebalance object
+	return o.rebalanceObject(karmadaClient, obj)
+}
+
+// getObjInfo get obj info in member cluster
+func (o *CommandRebalanceOption) getObjInfo(f cmdutil.Factory, args []string) (*unstructured.Unstructured, error) {
+	r := f.NewBuilder().
+		Unstructured().
+		NamespaceParam(o.Namespace).
+		FilenameParam(false, &o.FilenameOptions).
+		RequestChunksOf(500).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
+
+	r.IgnoreErrors(apierrors.IsNotFound)
+
+	infos, err := r.Infos()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(infos) == 0 {
+		return nil, fmt.Errorf("the %s or %s don't exist in Karmada", args[0], args[1])
+	}
+
+	return infos[0].Object.(*unstructured.Unstructured), nil
+}
+
+func (o *CommandRebalanceOption) rebalanceObject(cli karmadaclientset.Interface, obj *unstructured.Unstructured) error {
+	if o.DryRun {
+		fmt.Printf("try to rebalance resource %s %s %s\n", obj.GetKind(), obj.GetNamespace(), obj.GetName())
+		return nil
+	}
+
+	rebalancer := &appsv1alpha1.WorkloadRebalancer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      string(uuid.NewUUID()),
+			Namespace: obj.GetNamespace(),
+			Labels:    map[string]string{rebalancerByCtl: "true"},
+		},
+		Spec: appsv1alpha1.WorkloadRebalancerSpec{
+			Workloads: []appsv1alpha1.ObjectReference{{
+				APIVersion: obj.GetAPIVersion(),
+				Kind:       obj.GetKind(),
+				Name:       obj.GetName(),
+				Namespace:  obj.GetNamespace(),
+			}},
+			TTLSecondsAfterFinished: ptr.To[int32](30 * 60), // default ttl 30 minutes
+		},
+	}
+
+	_, err := cli.AppsV1alpha1().WorkloadRebalancers().Create(context.TODO(), rebalancer, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to trigger rebalance when creating WorkloadRebalancer: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to trigger rebalance: creating WorkloadRebalancer err: %w", err)
```

Reply:
The expression is problematic; it should be 'failed to create WorkloadRebalancer when triggering rebalancer' instead of 'failed to trigger rebalance when creating WorkloadRebalancer'.